pkg/operator/e2e/main_test.go

```diff
@@ -778,3 +788,63 @@ spec:
 		t.Fatalf("failed waiting for generated rules: %s", err)
 	}
 }
+
+func testValidateRuleEvaluationMetrics(ctx context.Context, t *testContext) {
+	// The project and cluster name in which we look for the metric data must
+	// be provided by the user. Check this only in this test so tests that don't need these
+	// flags can still be run without them.
+	if projectID == "" {
+		t.Fatalf("no project specified (--project-id flag)")
+	}
+	if cluster == "" {
+		t.Fatalf("no cluster name specified (--cluster flag)")
+	}
+
+	// Wait for metric data to show up in Cloud Monitoring.
+	metricClient, err := gcm.NewMetricClient(ctx)
+	if err != nil {
+		t.Fatalf("Create GCM metric client: %s", err)
+	}
+	defer metricClient.Close()
+
+	err = wait.Poll(1*time.Second, 3*time.Minute, func() (bool, error) {
+		now := time.Now()
+
+		// Validate the majority of labels being set correctly by filtering along them.
+		iter := metricClient.ListTimeSeries(ctx, &gcmpb.ListTimeSeriesRequest{
+			Name: fmt.Sprintf("projects/%s", projectID),
+			Filter: fmt.Sprintf(`
+				resource.type = "prometheus_target" AND
+				resource.labels.project_id = "%s" AND
+				resource.labels.cluster = "%s" AND
+				resource.labels.namespace = "%s" AND
+				metric.type = "prometheus.googleapis.com/always_one/gauge"
+				`,
+				projectID, cluster, t.namespace,
+			),
+			Interval: &gcmpb.TimeInterval{
+				EndTime:   timestamppb.New(now),
+				StartTime: timestamppb.New(now.Add(-10 * time.Second)),
+			},
+		})
+		series, err := iter.Next()
+		if err == iterator.Done {
+			t.Logf("No data, retrying...")
+			return false, nil
+		} else if err != nil {
+			return false, errors.Wrap(err, "querying metrics failed")
+		}
+		if len(series.Points) == 0 {
+			return false, errors.New("unexpected zero points in result series")
+		}
+		// We expect exactly one result.
```

Comment:
Why is this? I can't find where we set our rules scraping interval to 10s...

Reply:
This is for the series, not the points within it.

Reply:
Ah right.